### PR TITLE
Fixing EKS and performanceTracking Integration Tests

### PR DESCRIPTION
**Pull request by bhanuba**
# Description of the issue
Found out that few integration tests were failing. These tests used to be running successfully previously. 
-EKS Integration 
-performanceTrackingtest

# Description of changes
-Changed from heredoc syntax (<<POLICY) to jsonencode for fixing the EKS Integration test. 
-The validators path was changed but not redirected in main.tf. Fixed by redirecting to the correct path. 

# License
By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

# Tests
Fix for EKS integration test: https://github.com/aws/amazon-cloudwatch-agent/actions/runs/6673973151
Fix for performanceTrackingtest : https://github.com/aws/amazon-cloudwatch-agent/actions/runs/6648952606

Screenshot 2023-10-30 at 3 57 40 PM
Screenshot 2023-10-30 at 4 01 00 PM

### CODE REVIEW

**Comment by okankoAMZ** [comment]
Nit: The title of the PR can be named better

**Comment by okankoAMZ** [comment]
The performance test looks good but can you double check the dynamo database to see if it saved the test correctly (maybe a screenshot). Does this also work for traces?

**Comment by zhihonl** [comment]
Please rename the title of the PR. It is not descriptive enough. Also provide more details for `Description of the issue` and `Description of changes` sections. It does not tell the reviewers what the actual issue is and what you did to fix the issues.

File: generator/test_case_generator.go [hunk 1]
```diff
@@ -80,8 +80,8 @@ var testTypeToTestConfig = map[string][]testConfig{
 			testDir: "./test/acceptance",
 			targets: map[string]map[string]struct{}{"os": {"ubuntu-20.04": {}}},
 		},
-		// skipping FIPS test as the test cannot be verified
-		// neither ssh nor SSM works after a reboot once FIPS is enabled
+		//skipping FIPS test as the test cannot be verified
```
Comment (zhihonl) [inline generator/test_case_generator.go:83]: nit: likely your formatters but we don't need these changes

File: terraform/eks/daemon/emf/main.tf [hunk 2]
```diff
@@ -63,21 +63,18 @@ resource "aws_eks_node_group" "this" {
...
-          "Effect": "Allow",
-          "Principal": {
-            "Service": "ec2.amazonaws.com"
-          },
-          "Action": "sts:AssumeRole"
-        }
-      ]
-    }
-    POLICY
+  assume_role_policy = jsonencode({
```
Comment (zhihonl) [inline terraform/eks/daemon/emf/main.tf:66]: Description of the PR isn't clear so can you explain what was the reasons for this change

File: terraform/performance/main.tf [hunk 3]
```diff
@@ -113,9 +113,11 @@ resource "null_resource" "validator_linux" {
       "git clone --branch ${var.github_test_repo_branch} ${var.github_test_repo}",
       var.run_mock_server ? "cd mockserver && sudo docker build -t mockserver . && cd .." : "echo skipping mock server build",
       var.run_mock_server ? "sudo docker run --name mockserver -d -p 8080:8080 -p 443:443  mockserver" : "echo skipping mock server run",
-      "cd ..", # return to root , two copy xray configs next to validator
       "cp -r amazon-cloudwatch-agent-test/test/xray/resources /home/ec2-user/",
       "export AWS_REGION=${var.region}",
+      "cd ./validator/validators",
+      "ls -lirt",
```
Comment (zhihonl) [inline terraform/performance/main.tf:119]: nit: Do we need `ls -lirt` and `pwd` command here?

File: test/metric_value_benchmark/metrics_value_benchmark_test.go [hunk 4]
```diff
@@ -129,6 +129,7 @@ func (suite *MetricBenchmarkTestSuite) TestAllInSuite() {
 	case computetype.EKS:
 		log.Println("Environment compute type is EKS")
 		for _, testRunner := range getEksTestRunners(env) {
+			fmt.Println("testRunner---------", testRunner.Env)
```
Comment (zhihonl) [inline test/metric_value_benchmark/metrics_value_benchmark_test.go:132]: nit: Is this print line needed?

**Review by zhihonl — APPROVED** [review]

**Pull request merged**